### PR TITLE
Allocate amr_loadbalance sendrecv bufs together

### PR DESCRIPTION
**Pull request by forrestglines**
## PR Summary

Allocate the `sendbuf` and `recvbuf`s for AMR communications inside `RedistributeAndRefineMeshBlocks` all at once.

Allocating the buffers together should speed up recomputing the AMR.

## PR Checklist

- [ ] Code passes cpplint
- [ ] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Code is formatted
- [ ] Changes are summarized in CHANGELOG.md
- [ ] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Comment by forrestglines** [comment]
@par-hermes format

**Comment by forrestglines** [comment]
This one is quick, ready for review. I'll add to the change log once the other Hackathon PRs are merged into `hackathon_integration` and I can cleanly rebase.

File: src/mesh/amr_loadbalance.cpp [hunk 1]
```diff
@@ -528,10 +528,62 @@ void Mesh::RedistributeAndRefineMeshBlocks(ParameterInput *pin, ApplicationInput
...
+        }
+      } else { // same level or c2f
+        if (ranklist[on] == Globals::my_rank) continue;
+        int size;
+        if (oloc.level == nloc.level) {
+          size = bssame;
+        } else {
+          size = bsc2f;
+        }
+        buf_size += size;
```
Comment (jlippuner) [inline src/mesh/amr_loadbalance.cpp:554]: Maybe make this a bit more succinct:
```suggestion
        buf_size += (oloc.level == nloc.level) ? bssame : bsc2f;
```

File: src/mesh/amr_loadbalance.cpp [hunk 2]
```diff
@@ -559,7 +613,9 @@ void Mesh::RedistributeAndRefineMeshBlocks(ParameterInput *pin, ApplicationInput
         } else {
           size = bsc2f;
         }
-        recvbuf[rb_idx] = BufArray1D<Real>("recvbuf" + std::to_string(rb_idx), size);
```
Comment (jlippuner) [inline src/mesh/amr_loadbalance.cpp:562]: This is style, but I'd use `int size = (oloc.level = nloc.level) ? bssame : bsc2f;` here

**Review by jlippuner — APPROVED** [review]
Looks good. I just have some small nitpicks. It would be nice to reduce some code duplication in this entire section, but that's probably beyond the scope of this PR

**Review by Yurlungur — APPROVED** [review]

**Pull request merged**